packages/web-ui/src/sections/Document/Sidebar/Files/index.tsx

```diff
@@ -276,41 +277,100 @@ function FileNode({
   )
 }
 
+type DeletableType = 'file' | 'folder'
+type DeletableElement<T extends DeletableType> = T extends 'file'
+  ? {
+      type: T
+      documentUuid: string
+      name: string
+    }
+  : {
+      type: T
+      path: string
+      name: string
+    }
+
 export function FilesTree({
   currentPath,
   documents,
   navigateToDocument,
   createFile,
   destroyFile,
+  destroyFolder,
+  isDestroying,
 }: {
   createFile: (args: { path: string }) => Promise<void>
   destroyFile: (documentUuid: string) => Promise<void>
+  destroyFolder: (path: string) => Promise<void>
   documents: SidebarDocument[]
   currentPath: string | undefined
   navigateToDocument: (documentUuid: string) => void
+  isDestroying: boolean
 }) {
   const togglePath = useOpenPaths((state) => state.togglePath)
   const rootNode = useTree({ documents })
+  const [deletableNode, setDeletable] =
+    useState<DeletableElement<DeletableType> | null>(null)
 
   useEffect(() => {
     if (currentPath) {
       togglePath(currentPath)
     }
   }, [currentPath, togglePath])
+  const onConfirmDelete = useCallback(
+    async <T extends DeletableType>(deletable: DeletableElement<T>) => {
+      if (deletable.type === 'file') {
```

Comment:
magic string

Reply:
Is a type if you put other thing Typescript fails

Reply:
ferran is wrong enums are easier to refactor and in practice you'll never hit any of the issues he mentions (you haven't had a problem with enums in 10 years of using them)